### PR TITLE
[UI Toolkit] Add color prop in typography components (Text) [LW-9760]

### PR DESCRIPTION
**Pull request by przemyslaw-wlodek**
# Checklist

- [x] [JIRA](https://input-output.atlassian.net/browse/LW-9760)
- [x] Screenshots added.

---

## Proposed solution
Introduce new prop `color` to our Typography components `<Text.*>` to remove redundant custom css styling.

## Testing
Should be mostly covered by Chromatic snapshots but please take a look at the real examples in the browser extension.

## Screenshots
image

### CODE REVIEW

File: packages/ui/src/design-system/profile-dropdown/profile-dropdown-wallet-card.component.tsx [hunk 1]
```diff
@@ -49,7 +49,7 @@ export const WalletCard = ({
         <UserProfile {...profile} radius="rounded" />
       )}
       <Flex flexDirection="column" ml="$10" h="$32" alignItems="flex-start">
-        <Title className={cx.title} data-testid={makeTestId(testId, '-title')}>
+        <Title color="secondary" data-testid={makeTestId(testId, '-title')}>
```
Comment (VanessaPC) [inline packages/ui/src/design-system/profile-dropdown/profile-dropdown-wallet-card.component.tsx:52]: as a further improvement, we could probably put the types of button in an enum and export it, 
`export enum ButtonType = { primary, secondary...}` 
not sure what the scope of this ticket was though.

File: packages/ui/src/design-system/action-card/action-card.component.tsx [hunk 2]
```diff
@@ -38,10 +38,7 @@ export const ActionCard = ({
           {title.map(({ text, highlight }) => (
             <Text.Body.Normal
               weight="$medium"
-              className={cn({
```
Comment (vetalcore) [inline packages/ui/src/design-system/action-card/action-card.component.tsx:41]: please remove an import of classnames lib

File: packages/ui/src/design-system/dialog/dialog-title.css.ts [hunk 3]
```diff
@@ -1,10 +1,6 @@
 import { style, sx } from '../../design-tokens';
 
 export const dialogTitle = style([
-  sx({
```
Comment (vetalcore) [inline packages/ui/src/design-system/dialog/dialog-title.css.ts:4]: ditto

File: packages/ui/src/design-system/file-upload/file-upload.component.tsx [hunk 4]
```diff
@@ -47,17 +47,14 @@ export const FileUpload = ({
           {labelText.map(({ text, highlight }) => (
             <Text.Body.Normal
               weight="$medium"
-              className={cn({
```
Comment (vetalcore) [inline packages/ui/src/design-system/file-upload/file-upload.component.tsx:50]: ditto

**Review by vetalcore — CHANGES_REQUESTED** [review]

**Review by vetalcore — APPROVED** [review]

File: packages/ui/src/design-system/profile-dropdown/profile-dropdown-wallet-card.component.tsx (hunk 1 above)
Comment (VanessaPC) [reply packages/ui/src/design-system/profile-dropdown/profile-dropdown-wallet-card.component.tsx:52]: After a huddle with @przemyslaw-wlodek I realised this is a text component, and that it's better to have the color exposed as it currently is, so ignore my suggestion.

File: packages/ui/.eslintrc.js [hunk 5]
```diff
@@ -125,6 +125,7 @@ module.exports = {
         allowMappedTypes: 'in-unions-and-intersections',
       },
     ],
+    '@typescript-eslint/no-unused-vars': 'error',
```
Comment (DominikGuzei) [inline packages/ui/.eslintrc.js:128]: can you give a little context about disabling this rule?

File: packages/ui/src/design-system/asset-input/ticker-button.component.tsx [hunk 6]
```diff
@@ -22,7 +22,7 @@ export const TickerButton = ({
     onClick={onClick}
     data-testid={`asset-input-ticker-button-${id}`}
   >
-    <Text.SubHeading weight="$bold" className={cx.ticker}>
+    <Text.SubHeading weight="$bold" color="primary">
```
Comment (tommayeliog) [inline packages/ui/src/design-system/asset-input/ticker-button.component.tsx:25]: @przemyslaw-wlodek isn't it color="primary" by default?

**Review by tommayeliog — CHANGES_REQUESTED** [review]

File: packages/ui/.eslintrc.js (hunk 5 above)
Comment (przemyslaw-wlodek) [reply packages/ui/.eslintrc.js:128]: I am enabling it 😄 It was just a warning before, and it doesn't stop me from committing this (commitlint gave thumbs up).

File: packages/ui/src/design-system/educational-card/educational-card-item.component.tsx [hunk 7]
```diff
@@ -32,11 +32,11 @@ export const Item = ({
     </Flex>
     <Box>
       <Box mb="$8">
-        <Text.Body.Normal weight="$medium" className={cx.label}>
+        <Text.Body.Normal weight="$medium" color="secondary">
           {label}
         </Text.Body.Normal>
       </Box>
-      <Text.Body.Large weight="$semibold" className={cx.title}>
+      <Text.Body.Large color="primary" weight="$semibold" className={cx.title}>
```
Comment (tommayeliog) [inline packages/ui/src/design-system/educational-card/educational-card-item.component.tsx:39]: @przemyslaw-wlodek color="primary" is by default

**Review by tommayeliog — CHANGES_REQUESTED** [review]

File: packages/ui/src/design-system/file-upload/file-upload.component.tsx [hunk 8]
```diff
@@ -67,7 +63,7 @@ export const FileUpload = ({
           {files.map((file, index) => (
             <Box key={file}>
               <Flex mb="$8">
-                <Text.Body.Small weight="$medium" className={cx.fileName}>
+                <Text.Body.Small color="primary" weight="$medium">
```
Comment (tommayeliog) [inline packages/ui/src/design-system/file-upload/file-upload.component.tsx:66]: color="primary" is by default

File: packages/ui/src/design-system/profile-dropdown/accounts/profile-dropdown-account-item.component.tsx [hunk 9]
```diff
@@ -45,13 +45,15 @@ export const AccountItem = ({
...
         <Text.Label
+          color="secondary"
           weight="$medium"
           className={cx.accountLabel}
           data-testid="wallet-account-item-label"
         >
           {label}
         </Text.Label>
         <Text.Body.Small
+          color="primary"
```
Comment (tommayeliog) [inline packages/ui/src/design-system/profile-dropdown/accounts/profile-dropdown-account-item.component.tsx:56]: color="primary" is by default

File: packages/ui/src/design-system/profile-dropdown/profile-dropdown-wallet-card.component.tsx [hunk 10]
```diff
@@ -59,8 +59,8 @@ export const WalletCard = ({
           })}
         >
           <Text.Body.Small
+            color="primary"
```
Comment (tommayeliog) [inline packages/ui/src/design-system/profile-dropdown/profile-dropdown-wallet-card.component.tsx:62]: color="primary" is by default

File: packages/ui/src/design-system/scroll-area/scroll-area.stories.tsx [hunk 11]
```diff
@@ -28,11 +27,9 @@ const Demo = (): JSX.Element => {
           <Cell>
             {Array.from({ length: 50 }).map((_, index) => (
               <Box my="$4" key={`tag:${index}`}>
-                <Text.Body.Normal
-                  className={sx({
-                    color: '$buttons_primary_label_color',
-                  })}
-                >{`v1.2.0-beta.${index + 1}`}</Text.Body.Normal>
+                <Text.Body.Normal color="primary">
```
Comment (tommayeliog) [inline packages/ui/src/design-system/scroll-area/scroll-area.stories.tsx:30]: color="primary" is by default

**Review by tommayeliog — CHANGES_REQUESTED** [review]

**Review by tommayeliog — APPROVED** [review]

File: packages/ui/.eslintrc.js (hunk 5 above)
Comment (DominikGuzei) [reply packages/ui/.eslintrc.js:128]: ah sorry, my bad :D

**Review by DominikGuzei — APPROVED** [review]

File: packages/ui/src/design-system/text/text.stories.tsx [hunk 12]
```diff
@@ -77,9 +79,33 @@ const typefaces = [
...
+          <Variants.Row>
+            {colors.map(color => (
+              <Variants.Cell key={color}>
+                <Body.Normal color={color}>Text</Body.Normal>
+              </Variants.Cell>
+            ))}
+          </Variants.Row>
+        </ColorSchemaTable>
+      </Section>
+      <Divider my="$64" />
```
Comment (tommayeliog) [inline packages/ui/src/design-system/text/text.stories.tsx:108]: "Typography" story in Storybook does not match Figma's design from the link in JIRA.
I don't know if Figma was update after you started working on this story.

Screenshot 2024-03-19 at 10 23 32

**Pull request merged**